.github/workflows/python-release-tests.yml

```diff
@@ -0,0 +1,87 @@
+name: Test Anemoi Packages for Release
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  quality:
+    uses: ecmwf/reusable-workflows/.github/workflows/qa-precommit-run.yml@v2
+    with:
+      skip-hooks: "no-commit-to-branch"
+
+  checks:
+    if: ${{ github.event_name == 'pull_request' && github.event.pull_request.user.login == 'DeployDuck'}}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ["ubuntu-latest", "macos-latest"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+    runs-on: ${{ matrix.platform }}
+    timeout-minutes: 60
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        fetch-tags: true
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+
+    - name: Install test dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest pytest-md pytest-emoji
+
+    - name: Detect package to be released
+      id: release-package
+      run: |
+        # For pull requests, compare with the PR base
+        MERGE_BASE=$(git merge-base origin/${{ github.event.pull_request.base.ref }} ${{ github.event.pull_request.head.sha }})
+        CHANGED_FILES=$(git diff --name-only $MERGE_BASE...${{ github.event.pull_request.head.sha }})
+
+        echo "Changed files:"
+        echo "$CHANGED_FILES"
+
+        echo "training_release=$(echo "$CHANGED_FILES" | grep -q '^training/release-please-manifest' && echo 'true' || echo 'false')" >> $GITHUB_OUTPUT
+
+        echo "graphs_release=$(echo "$CHANGED_FILES" | grep -q '^graphs/release-please-manifest' && echo 'true' || echo 'false')" >> $GITHUB_OUTPUT
+
+        echo "models_release=$(echo "$CHANGED_FILES" | grep -q '^models/release-please-manifest' && echo 'true' || echo 'false')" >> $GITHUB_OUTPUT
+
+
+    - name: Install packages
+      if: ${{steps.release-package.outputs.training_release == 'true' || steps.release-package.outputs.graphs_release == 'true' || steps.release-package.outputs.models_release == 'true' }}
+      run: |
+        if [[ ${{ steps.release-package.outputs.training_changed }} == 'true' ]]; then
+          pip install -e ./training[all,tests]
+        elif [[ ${{ steps.release-package.outputs.graphs_release }} == 'true' ]]; then
+          pip install -e ./graphs[all,tests]
+        elif [[ ${{ steps.release-package.outputs.models_release }} == 'true' ]]; then
```

Comment:
Did you use the correct variable names here? Should this be `training_release` instead of `training_changed` ?